src/Microsoft.FeatureManagement/FeatureManagementBuilderExtensions.cs

```diff
@@ -18,7 +20,16 @@ public static class FeatureManagementBuilderExtensions
         /// <returns>A <see cref="IFeatureManagementBuilder"/> that can be used to customize feature management functionality.</returns>
         public static IFeatureManagementBuilder WithTargeting<T>(this IFeatureManagementBuilder builder) where T : ITargetingContextAccessor
         {
-            builder.Services.TryAddSingleton(typeof(ITargetingContextAccessor), typeof(T));
+            //
+            // Register the targeting context accessor with the same lifetime as the feature manager
+            if (!builder.Services.Any(descriptor => descriptor.ServiceType == typeof(FeatureManager) && descriptor.Lifetime == ServiceLifetime.Scoped))
```

Comment:
This condition seems backwards. Addsingleton if feature manager is registered as scoped?

Reply:
Oh it's negated.

Reply:
```suggestion
            if (builder.Services.Any(descriptor => descriptor.ServiceType == typeof(FeatureManager) && descriptor.Lifetime == ServiceLifetime.Scoped))
            {
                // register as scoped
            }
```

I feel it will be clearer to remove the negation and instead override to scoped if you do detect a scoped feature manager.